### PR TITLE
adding the ability to edit usernames

### PR DESCRIPTION
**Pull request by soonahn**
Addresses https://github.com/RocketChat/Rocket.Chat/issues/695

### CODE REVIEW

**Comment by soonahn** [comment]
I also removed a bunch of redspace near where I was working.  Long empty lines make me cringe to see, can't help but remove.

**Comment by geekgonecrazy** [comment]
Just a little UI request.  Maybe order Username / Password / Language ?

**Comment by soonahn** [comment]
Sure thing

**Comment by soonahn** [comment]
Added the username as a placeholder.

**Comment by geekgonecrazy** [comment]
Looks great!  Thank you!

**Comment by engelgabriel** [comment]
Guys, update the username is not that simple. You need to use the method. There are a lot of checking and references that need o be updated together, not just the users collection.

**Comment by engelgabriel** [comment]
Need to call /packages/rocketchat-lib/server/methods/setUsername.coffee

**Comment by soonahn** [comment]
Setting the username properly.  Hopefully good enough!

**Comment by soonahn** [comment]
@engelgabriel updated, anything else missing?

**Comment by geekgonecrazy** [comment]
@soonahn Sorry for the confusion.  I thought I had mentioned the use of `Meteor.call 'setUsername'` method when we initially talked.

Tested this out.  (this time with actual messages to make sure those are still changed)  Sorry oversight last time.  :)

LGTM i'll leave this for @engelgabriel though.  :)

File: packages/rocketchat-lib/server/methods/setUsername.coffee [hunk 1]
```diff
@@ -11,7 +11,7 @@ Meteor.methods
 			return username
 
 		if not /^[0-9a-zA-Z-_.]+$/.test username
-			throw new Meteor.Error 'username-invalid'
+			throw new Meteor.Error 'username-invalid', TAPi18next.t('project:Username_invalid', username)
```
Comment (engelgabriel) [inline packages/rocketchat-lib/server/methods/setUsername.coffee]: Why `TAPi18next.t('project:Username_invalid', username)`?
The translation should not be done on the server side, as the client may be using a different language.

Comment (soonahn) [inline packages/rocketchat-lib/server/methods/setUsername.coffee]: Still new to Meteor, unsure which pieces fit where. I assumed the client data got passed down and translated then passed back up to the server.

Sorry!  I'll read the docs better next time.

**Pull request merged**